### PR TITLE

Enable entry actions when editing an entry and cleanup entry context menu

### LINKED ISSUE #2118: Add copy password button to Edit Entry dialog
In KeepassX if I open an entry for editing, say to add the URL, I can still right-click and copy the password without exposing it. In XC I'm not able to do that. I either have to close the edit window or expose the password. 

## Expected Behavior
right-click menu in Edit Entry dialog should include "copy password" (and "copy username" tbh). So should the preview/overview of the key. 

## Current Behavior
The right-click menu only includes undo/redo, cut/copy/paste/delete and select all. If I select a password that's hidden, all but "delete" and "select all" are greyed out. 

If I don't open a key, there's no right click behaviour at all on the bottom half of the screen. I can select the username and right-click to copy it, but I have to right click on the entry in the list above the overview/preview pane to copy the password. 

## Possible Solution
Add "copy username" and "copy password" to the context menu in the Edit Entry dialog and the overview/preview of each entry.
 
The preview pane should include the full context menu that's available in the search results. 

## Context
I often want to view the details of an entry to confirm that it's the one I'm looking for, so I double click or <kbd>enter</kbd> after a search, which brings up the edit window. Once I'm in the edit window if I want to copy the password over without showing it to my colleagues who are all unrepentant shoulder surfers, I have to <kbd>cancel</kbd> and then use the "copy password" option in the right click context menu. It would be really helpful if it was available anytime I'm viewing or editing an entry.  

I've been using KeepassXC for a few weeks and I thought I'd eventually get used to it but it's absurd how often I have to take that extra step to get the password into my clipboard without showing it to the world. 

## Debug Info
KeePassXC - Version 2.3.3
Revision: 0a155d8

Libraries:
- Qt 5.9.5
- libgcrypt 1.8.1

Operating system: Ubuntu 18.04 LTS
CPU architecture: x86_64
Kernel: linux 4.15.0-24-generic

Enabled extensions:
- Auto-Type
- Browser Integration
- Legacy Browser Integration (KeePassHTTP)
- SSH Agent
- YubiKey

### LINKED ISSUE #2813: Order of context menu options
KeePassXC 2.4.0 for Windows has changed the order of the context menu options. Instead of having the most used options at the top (`Copy username` and `password`), now it has `New Entry`, even when you're clicking on an existing entry. 

Please consider placing those two options at the top again or letting us customize the order.

### PR DESCRIPTION
**Pull request by droidmonkey**
[TIP]:  # ( Provide a general summary of your changes in the title above ^^ )

## Type of change
[NOTE]: # ( Please remove all lines which don't apply. )
- ✅ Refactor (significant modification to existing code)
- ✅ New feature (non-breaking change which adds functionality)

## Description and Context
[NOTE]: # ( Describe your changes in detail, why is this change required? )
[NOTE]: # ( Describe the context of your change. Explain large code modifications. )
[NOTE]: # ( If it fixes an open issue, please add "Fixes #XXX" as necessary )

**Enable entry actions when editing an entry**
* Fixes #2118 

* Enables select entry actions when editing an entry. This allows users to copy the password of the entry, for example.
* _Note: unsaved changes to the entry will not be copied to the clipboard._

**Add special context menu for entries**
* Fixes #2813

* Added special context menu for entries to move the most-used actions to the top.
* Re-ordered actions in the entry menu to be more user friendly.

## Screenshots
[TIP]:  # ( Do not include screenshots of your actual database! )
image

image

image

## Testing strategy
[NOTE]: # ( Please describe in detail how you tested your changes. )
[TIP]:  # ( We expect new code to be covered by unit tests and documented with doc blocks! )
Tested manually

## Checklist:
[NOTE]: # ( Please go over all the following points. )
[NOTE]: # ( Again, remove any lines which don't apply. )
[NOTE]: # ( Pull Requests that don't fulfill all [REQUIRED] requisites are likely )
[NOTE]: # ( to be sent back to you for correction or will be rejected.  )
- ✅ I have read the **CONTRIBUTING** document. **[REQUIRED]**
- ✅ My code follows the code style of this project. **[REQUIRED]**
- ✅ All new and existing tests passed. **[REQUIRED]**
- ✅ I have compiled and verified my code with `-DWITH_ASAN=ON`. **[REQUIRED]**
- ✅ My change requires a change to the documentation, and I have updated it accordingly.
- ✅ I have added tests to cover my changes.

### CODE REVIEW

File: src/gui/DatabaseWidget.cpp [hunk 1]
```diff
@@ -751,6 +756,15 @@ void DatabaseWidget::openUrlForEntry(Entry* entry)
...
 }
 
+Entry* DatabaseWidget::currentSelectedEntry()
+{
+    if (currentWidget() == m_editEntryWidget) {
+        return m_editEntryWidget->currentEntry();
+    } else {
+        return m_entryView->currentEntry();
+    }
+}
```
Comment (phoerious) [inline src/gui/DatabaseWidget.cpp]: Redundant else.

Comment (droidmonkey) [reply src/gui/DatabaseWidget.cpp]: Agreed

**Comment by FlorentDotMe** [comment]
I fear this PR introduced unexpected behavior. I agree with what is mentioned in #2813, but when no entry is selected, the standard context menu isn't available anymore. I don't think that was the point of this PR.

So we can't add new entries anymore by right clicking in a group view.

### Current behavior
KeePassXC_Menu_Current01

### Expected behavior
KeePassXC_Menu_Suggestion01

**Comment by droidmonkey** [comment]
The toolbar has that option permanently visible.

**Comment by FlorentDotMe** [comment]
If you consider the second custom menu as the most appropriate option, so that's the one ;) But anyway, keeping the "New Entry" in the mouse menu, ideally when no entry is selected (if too much customization, better to revert back to always visible?), seems essential to me. Thanks!

**Pull request merged**